File: api/src/controllers/get.js

const { Model } = require('sequelize');
const service = require('../models/service');
const User = require('../models/user');

class GetController {
    async getServices(req, res) {
        try {
            // Busca todos os serviços, incluindo as informações do usuário associado
            const services = await Service.findAll({
                include: [{
                    model: User,
                    as: 'usuarios', 
                    attributes: ['nome']
                }]
            });

            // Retorna os serviços encontrados
            res.status(200).json(services);
        } catch (error) {
            console.error('Erro ao buscar serviços:', error);
            res.status(500).json({ error: 'Erro ao buscar serviços', details: error.message });
        }
    }
}


module.exports = getController;

File: api/src/controllers/get.go
package handlers

import (
	"api/src/models"
	"database/sql"
	"encoding/json"
	"net/http"
)

func GetUsers(w http.ResponseWriter, r *http.Request, db *sql.DB) {
	user, ok := r.Context().Value("user").(models.User)
	if !ok {
		http.Error(w, "Usuário não autenticado", http.StatusUnauthorized)
		return
	}
	users, err := models.GetUsersAll(db)
	if err != nil {
		http.Error(w, "Erro ao buscar usuários", http.StatusInternalServerError)
		return
	}
	w.Header().Set("Content-Type", "application/json")
	json.NewEncoder(w).Encode(map[string]interface{}{
		"message": "Lista de usuários recuperada com sucesso!",
		"user":    user,
		"users":   users,
	})
}

func GetServices(w http.ResponseWriter, r *http.Request, db *sql.DB) {
	service, ok := r.Context().Value("service").(models.Services)
	print(service.Servico)
	if !ok {
		http.Error(w, "Serviços não encontrados", http.StatusNotFound)
		return
	}
	services, err := models.GetServicesAll(db)
	if err != nil {
		http.Error(w, "Erro ao buscar serviços", http.StatusInternalServerError)
		return
	}
	w.Header().Set("Content-Type", "application/json")
	json.NewEncoder(w).Encode(map[string]interface{}{
		"message":  "Lista de serviços recuperada com sucesso!",
		"service":  service,
		"services": services,
	})
}
